### ISSUE TITLE
Add static labels

### ISSUE THREAD

**Issue by sandersaares** [prometheus-net/prometheus-net#256]
This seems to be a popular ask over the years, so perhaps it is worth adding to the library. Static labels with predefined values that never change.

This would be on two levels: registry and metric.

* Registry-level static labels would apply to all metrics in that registry.
* Metric-level static labels would apply to all instances of that metric.

Static labels cannot be overridden (at least in first version). If an attempt is made to define a label with a name that is already used for a static label, an exception is thrown.

These labels would be creatable either via Metrics.CreateXXX() or somehow directly on the registry. There would not be helpers to add them only to Http/HttpClient/gRPC metrics (since it seems likely this scenario can be satisfied by global registry-level labels), though you could still make that happen by defining custom metrics with the desired static labels.

**Comment by sandersaares** [comment]
API draft:

```csharp
Metrics.DefaultRegistry.StaticLabels.Add("foo", "bar");
foreach (KeyValuePair<string, string> label in Metrics.DefaultRegistry.StaticLabels) { ... }

var gauge = Metrics.CreateGauge("foo", "help", new GaugeConfiguration
{
    StaticLabels = new[]
    {
        new KeyValuePair<string, string>("foo", "bar")
    }
});

// With convenience helper for less typing in language versions without target-typed new.
var gauge = Metrics.CreateGauge("foo", "help", new GaugeConfiguration
{
    StaticLabels = StaticLabels.Create(
        "foo", "bar",
        "baz", "wololo")
});
```

**Comment by sandersaares** [comment]
Now available in prerelease build `4.1.0-pre-000171-15be8f3`: https://github.com/prometheus-net/prometheus-net#static-labels

**Comment by DuckyC** [comment]
Is there any reason why these values cannot be changed at runtime?
